http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java

```diff
@@ -0,0 +1,163 @@
+            [{description=Send a request with a YAML payload to the endpoint, including cities data, steps=[{action=POST, endpoint=/media-type/yaml, requestHeaders={Content-Type=application/yaml}, requestBody={cities=[{name=Tokio, country=Japan}]}}], expectedResponse={status=200, headers={Content-Type=application/json}, body={cities=[{name=Tokio, country=Japan}]}}}]""";
+
+    @Test
+    public void testExpectedXmlResponse() {
+        app.given()
+                .get("/city").then()
+                .statusCode(HttpStatus.SC_OK)
+                .contentType(MediaType.APPLICATION_XML)
+                .body("cityListWrapper.cities[0].name", equalTo("San Bernardino"))
+                .body("cityListWrapper.cities[0].country", equalTo("EEUU"));
+
+    }
+
+    @Test
+    public void testXMLPayloadPostRequest() throws JAXBException {
+
+        City city = new City("Paris", "France");
+        List<City> cityList = new ArrayList<>();
+        cityList.add(city);
+
+        CityListDTO requestCityList = new CityListDTO(cityList);
+
+        String payload = app.given().contentType(MediaType.APPLICATION_XML)
+                .body(requestCityList) // Use the generated XML payload
+                .when()
+                .post("/city")
+                .then()
+                .extract().asString();
+        Document doc = Jsoup.parse(payload);
+        String unescapedXml = doc.text();
+        assertThat(unescapedXml, equalTo(EXPECTED_XML_PAYLOAD));
+    }
+
+    @Test
+    public void testExpectedYamlResponse() {
+
+        final Response response = app.given()
+                .contentType("application/yaml")
+                .get("/city/getYamlFile")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract().response();
+
+        assertThat(response.asString(), containsStringIgnoringCase(EXPECTED_RESPONSE_BODY_YAML));
+
+    }
+
+    @Test
+    public void testPostYamlPayloadRequest() throws IOException {
+
+        String yamlString = readYamlFile("src/test/resources/payload.yaml");
+
+        Response response = RestAssured
+                .given()
+                .config(RestAssured.config()
+                        .encoderConfig(encoderConfig().encodeContentTypeAs("application/x-yaml", ContentType.TEXT)))
+                .contentType("application/x-yaml")
+                .body(yamlString)
+                .when()
+                .post("/city/cityYaml")
+                .then()
+                .statusCode(200).extract().response();
+        JsonPath jsonPath = response.jsonPath();
+
+        List<Object> cities = jsonPath.getList("expectedResponse.body.cities");
+        assertEquals("[{name=Tokio, country=Japan}]", cities.get(0).toString());
+    }
+
+    @Test
+    public void testGetImage() {
+        byte[] imageData = RestAssured.given().get("/city/wolf_moon_howling.jpg").asByteArray();
+        assertThat(imageData, notNullValue());
```

Comment:
I apologize for my gruelling review, but I'm worried this could be true even when byte array is empty? Anyway, only way to assure it works properly is to compare actual byte arrays.

Maybe I am missing something, please let me know if you disagree.

Reply:
ha! exactly in a fashion you are doing it in the next test would be great!